cli/src/main/java/com/devonfw/tools/ide/tool/IdeasyCommandlet.java

```
@@ -106,6 +106,11 @@ public Path getToolPath() {
   @Override
   public boolean install(boolean silent) {
 
+    if (!this.context.isOnline()) {
+      this.context.warning("You are offline. IDEasy requires an internet connection to upgrade - skipping upgrade.");
+      return false;
+    }
+
```

```suggestion
this.context.requireOnline("upgrade of IDEasy");
```

---

We should throw a `CliOfflineException` instead of logging a warning and ending with success if there was actually an error.
As the name of the method `requireOnline` suggests that I am required to be online, I was wrong with my suggestion.
Thanks for pointing that out.
Since that method is already used in other places we cannot simply change its implementation accordingly.
However, it would be smart to add a new optional boolean parameter to `requireOnline` that allows us to specify if we want to trigger an explicit `isOnline()` check or keep the existing behaviour.

---

I’ve implemented the change by adding a boolean parameter `explicitOnlineCheck` to `requireOnline()` that allows triggering an explicit `isOffline()` check, while keeping the existing behavior as default. I believe this reflects what was suggested, but let me know if I misunderstood anything.

I used `isOffline()` instead of `!isOnline()` to ensure it still works correctly in offline-mode.

Also updated the test to assert for the exception instead of checking for a warning log.